### PR TITLE
Annotations lookup fix

### PR DESCRIPTION
**Pull request by wesboyt**
*Issue #, if available:*
#513
*Description of changes:*
Fixes a few issues with symbol resolution within both text and binary.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/IonParserTextRaw.ts [hunk 1]
```diff
@@ -1079,6 +1079,12 @@ export class ParserTextRaw {
         ch = this._read_after_whitespace(true);
         if (ch == CH_CL && this._peek() == CH_CL) {
             this._read(); // consume the colon character
+            if(s[0] === '$'[0]) {
```
Comment (pbcornell) [inline src/IonParserTextRaw.ts]: simplify to: `if (s[0] === '$') {`

File: src/IonBinaryReader.ts [hunk 2]
```diff
@@ -249,9 +249,14 @@ export class BinaryReader implements Reader {
     private getSymbolString(symbolId: number): string {
         let s: string = null;
         if (symbolId > 0) {
+            if(symbolId > this._symtab.maxId) {
+                throw new Error('Symbol $' + symbolId.toString() + ' greater than maxID.');
+            }
             s = this._symtab.getSymbolText(symbolId);
-            if (typeof (s) == 'undefined') {
-                s = "$" + symbolId.toString();
+            if (typeof (s) === 'undefined') {
```
Comment (pbcornell) [inline src/IonBinaryReader.ts]: not sure why we're using `typeof` here;  consider changing to `if (s === undefined) {`

File: src/IonBinaryReader.ts [hunk 3]
```diff
@@ -249,9 +249,14 @@ export class BinaryReader implements Reader {
     private getSymbolString(symbolId: number): string {
         let s: string = null;
         if (symbolId > 0) {
+            if(symbolId > this._symtab.maxId) {
```
Comment (pbcornell) [inline src/IonBinaryReader.ts]: add a space after `if` please

File: src/IonTextReader.ts [hunk 4]
```diff
@@ -214,7 +214,21 @@ export class TextReader implements Reader {
     }
 
     annotations(): string[] {
-        return this._parser.annotations();
+        let ann = [];
+        for (let str of this._parser.annotations()) {
+            ann.push(str);
+        }
+        for (let i = 0; i < ann.length; i++) {
+            if(ann[i].length > 1 && ann[i].charAt(0) === '$'.charAt(0)) {
```
Comment (pbcornell) [inline src/IonTextReader.ts]: simplify to: `if (ann[i].length > 1 && ann[i][0] === '$') {`

File: src/IonTextReader.ts [hunk 5]
```diff
@@ -214,7 +214,21 @@ export class TextReader implements Reader {
...
+                let tempStr = ann[i].substr(1, ann[i].length);
+                if (+tempStr === +tempStr) {//look up sid, +str === +str is a one line is integer hack
+                    let symbol = this._symtab.getSymbolText(Number(tempStr));
+                    if(symbol === undefined || symbol === null) throw new Error("Unresolveable symbol ID, symboltokens unsupported.");
```
Comment (pbcornell) [inline src/IonTextReader.ts]: "Unresolveable" --> "Unresolvable"

**Comment by wesboyt** [comment]
> Minor comments, and one question:
> 
> * why are we adding `ion-tests/iontestdata/good/item1.10n` to the good/event skiplists?

Because it uses a shared symbol table and now throws errors, we are currently propagating symbols that can never be resolved silently.

File: src/IonBinaryReader.ts (hunk 2 above)
Comment (wesboyt) [reply src/IonBinaryReader.ts]: me either

**Pull request merged**